presto-druid/src/main/java/com/facebook/presto/druid/DruidBrokerPageSource.java

```diff
@@ -144,11 +141,8 @@ else if (type instanceof RealType) {
                             type.writeLong(blockBuilder, floatToRawIntBits(value.floatValue()));
                         }
                         else if (type instanceof TimestampType) {
-                            DateTimeFormatter formatter = ISODateTimeFormat.dateTimeParser()
-                                    .withChronology(ISOChronology.getInstanceUTC())
-                                    .withOffsetParsed();
-                            DateTime dateTime = formatter.parseDateTime(value.textValue());
-                            type.writeLong(blockBuilder, dateTime.getMillis());
+                            long millis = Instant.parse(value.textValue()).toEpochMilli();
```

Comment:
`Instant.parse` assumes the string to be in [DateTimeFormatter.ISO_INSTANT](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_INSTANT) format
https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#parse-java.lang.CharSequence-
Can you check if this equivalent to the joda code on the left.?
Relevant SO link : https://stackoverflow.com/questions/40727884/java8-equivalent-of-jodas-isodatetimeformat-datetimeparser

Reply:
there are cases when parsing fails.
you can play around with the different formats.
In the example below, joda is lenient with dropping the `Z`, but java complains.

```
@DataProvider
    public Object[][] dateStrings()
    {
        return new Object[][]
                {
                        {"2007-12-03T10:15:30.00"},
                        {"2007-12-03T10:15:30.00Z"}
                };
    }

    @Test(dataProvider = "dateStrings")
    public void testParsingCompatibility(String dateString)
    {
        DateTimeFormatter jodaFormatter = ISODateTimeFormat.dateTimeParser()
                .withChronology(ISOChronology.getInstanceUTC())
                .withOffsetParsed();
        
        assertEquals(
                jodaFormatter.parseDateTime(dateString).getMillis(),
                Instant.parse(dateString).toEpochMilli());
    }
  ```

Reply:
OK, let me look at this case and see if there might be others. I think this is a bug in JodaTime but it could be a bug we depend on.